### DISCUSSION TITLE
Does SqsListener always poll AWS queue in batches?

### DISCUSSION THREAD

**Discussion by kat-lgtm** [awspring/spring-cloud-aws#1172] [Q&A]
Hi there, 

We have been facing some issues with AWS throttling our delete requests for messages using a FIFO queue, and wanted to understand exactly how many send, receive and delete calls we are making when we use `@SqsListener`. 

Say there are 100 messages in a FIFO SQS queue (all in different message groups). 

1. If we use `@SqsListener` to receive single messages (with all default configurations and maxMessagesPerPoll=10), under the hood will it be polling the AWS queue / using "receive message" operation 10 times, or 100 times? 

2. If we use `@SqsListener` to receive a batch (also with defaults and maxMessagesPerPoll=10), since it's a FIFO queue and the messages are in different message groups - we will receive the messages 100 separate times from SqsListener. However, in this case will the queue only be polled 10 times?

Thank you for the clarification!

**Comment by maciejwalkowiak** [comment]
SQS does not guarantee the exact messages per poll, just the max messages per poll. So if you have 100 messages in queue and have max messages per poll set to 10, the minimum number of polls to SQS is equal 10 but can be more.
